.github/workflows/ci.yml

```diff
@@ -0,0 +1,575 @@
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Rust Info
+        uses: ./.github/rust-info
+
+      - name: Codecov
+        env:
+          # For codecov it's sufficient to run the fuzz tests only once.
+          QUICKCHECK_TESTS:              1
+          INK_COVERAGE_REPORTING:        true
+          CARGO_INCREMENTAL:             0
+          # Needed because `codecov` requires nightly features to work
+          # (see `-Z` in the `RUSTFLAGS` below).
+          RUSTC_BOOTSTRAP:               1
+          # Variables partly came from https://github.com/mozilla/grcov/blob/master/README.md
+          RUSTFLAGS:                     -Zprofile -Zmir-opt-level=0 -Ccodegen-units=1
+                                          -Clink-dead-code -Copt-level=0 -Coverflow-checks=off
+          # The `cargo-taurpalin` coverage reporting tool seems to have better code instrumentation and thus
+          # produces better results for Rust codebases in general. However, unlike `grcov` it requires
+          # running docker with `--security-opt seccomp=unconfined` which is why we use `grcov` instead.
+        run: |
+          # RUSTFLAGS are the cause target cache can't be used here
+          # FIXME: cust-covfix doesn't support the external target dir
+          # https://github.com/Kogia-sima/rust-covfix/issues/7
+          unset "CARGO_TARGET_DIR"
+          cargo clean
+          # make sure there's no stale coverage artifacts
+          find . -name "*.profraw" -type f -delete
+          find . -name "*.gcda" -type f -delete
+          # RUSTFLAGS are the cause target cache can't be used here
+          cargo build --verbose --all-features --workspace
+          cargo test --verbose --all-features --no-fail-fast --workspace
+          # coverage with branches
+          grcov . --binary-path ./target/debug/ --source-dir . --output-type lcov --llvm --branch \
+            --ignore-not-existing --ignore "/*" --ignore "tests/*" --output-path lcov-w-branch.info
+          rust-covfix lcov-w-branch.info --output lcov-w-branch-fixed.info
+          # TODO: Enable when Gitlab CI is disabled
+          #codecov --token "$CODECOV_P_TOKEN" --file lcov-w-branch-fixed.info --nonZero
+          # lines coverage
+          grcov . --binary-path ./target/debug/ --source-dir . --output-type lcov --llvm \
+            --ignore-not-existing --ignore "/*" --ignore "tests/*" --output-path lcov-lines.info
+          rust-covfix lcov-lines.info --output lcov-lines-fixed.info
+          # TODO: Enable when Gitlab CI is disabled
+          #codecov --token "$CODECOV_TOKEN" --file lcov-lines-fixed.info --nonZero
+
+### examples
+
+  examples-test:
+    runs-on: ubuntu-latest-8-cores
+    needs: [clippy]
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: paritytech/ci-unified:bullseye-1.73.0
```

Comment:
```suggestion
      image: ${{ env.IMAGE }}
```

Reply:
I think that in this context, environment variables are not being handled